GameLogic/entities/Move.cs

```diff
@@ -3,17 +3,22 @@
 namespace GameLogic
 {
     // The structure defines move. Each move can
-    // be uniquely identified by pair of start and end cells.
+    // be uniquely identified by a triplet of start cell, end cell, and end piece type.
+    // For example, the pawn can be promoted to another piece. Therefore, the piece type is required.
     public struct Move
     {
         public readonly Cell StartCell { get; }
         public readonly Cell EndCell { get; }
+        public readonly PieceType? EndPieceType { get; }
```

Comment:
I believe `End` is redundant here as it adds no value.

I suggest naming it `PromotionPieceType` or even shorter `PromotionPiece`.

Reply:
Fixed!